src/layouts/InterfaceLayout/containers/InteractWithContractContainer/InteractWithContractContainer.vue

```diff
@@ -1,84 +1,349 @@
+      resType: 'string',
+      gasLimit: 21000,
+      value: 0,
+      data: '',
+      raw: {},
+      nonce: 0,
+      signedTx: '',
+      transactionFee: 0
+    }
+  },
+  methods: {
+    selectedCurrency (currency) {
+      if (currency.abi === '') {
+        this.abi = ''
+      } else {
+        this.abi = JSON.stringify(currency.abi)
+      }
+      this.address = currency.address
+    },
+    checkType (type) {
+      if (type === 'address' || type === 'string' || type === 'byte') {
+        return 'string'
+      } else {
+        return 'number'
+      }
+    },
+    selectFunction (method) {
+      const contract = new this.$store.state.web3.eth.Contract(
+        [method],
+        this.address
+      )
+      if (method.constant === true && method.inputs.length === 0) {
+        contract.methods[method.name]()
+          .call({ from: this.$store.state.wallet.getAddressString() })
+          .then(res => {
+            this.result = res
+          })
+          .catch(err => {
+            console.log(err)
+          })
+      } else {
+        this.result = ''
+      }
+      this.selectedMethod = method
+    },
+    copyToClipboard (ref) {
+      this.$refs[ref].select()
+      document.execCommand('copy')
+    },
+    deleteInput (ref) {
+      this.$refs[ref].value = ''
+    },
+    switchView (direction) {
+      switch (direction) {
+        case 'forward':
+          this.methods = JSON.parse(this.abi)
+            .map(func => func)
+            .filter(func => func.type !== 'constructor')
+            .filter(func => func.constant !== undefined)
+          this.interact = true
+          break
+        default:
+          this.interact = false
+      }
+    },
+    async write () {
+      const contract = new this.$store.state.web3.eth.Contract([this.selectedMethod], this.address)
+      const params = Object.keys(this.writeInputs).map(input => this.writeInputs[input])
+      this.loading = true
+      if (this.selectedMethod.constant === true) {
+        contract.methods[this.selectedMethod.name](...params)
+          .call({ from: this.$store.state.wallet.getAddressString() })
+          .then(res => {
+            this.result = res
+            this.loading = false
+          })
+          .catch(err => {
+            console.log(err)
+            this.loading = false
+          })
+      } else {
+        this.nonce = await this.$store.state.web3.eth.getTransactionCount(this.$store.state.wallet.getAddressString())
+        this.gasLimit = await contract.methods[this.selectedMethod.name](
+          ...params
+        )
+          .estimateGas({ from: this.$store.state.wallet.getAddressString() })
+          .then(res => {
+            this.transactionFee = unit.fromWei(unit.toWei(this.$store.state.gasPrice, 'gwei') * res, 'ether')
+            return res
+          })
+          .catch(err => console.log(err))
+        this.data = contract.methods[this.selectedMethod.name](...params).encodeABI()
```

Comment:
you can use 
```
contract.methods[this.selectedMethod.name](...params).send({ from: this.$store.state.wallet.getAddressString() gasPrice:'0x...', gas: '0x...', value:'0x00'})
``` 

no need to create the tx manually

Reply:
Same error as mentioned above.